src/main/java/org/opensrp/repository/postgres/SettingRepositoryImpl.java

```diff
@@ -520,7 +533,68 @@ private void checkWhetherMetadataExistsBeforeSave(SettingConfiguration entity, L
 				settingsMetadataList.add(metadata);
 			}
 		}
-			settingMetadataMapper.insertMany(settingsMetadataList);
+
+		insertSettingMetadata(settingsMetadataList);
+	}
+
+	private void insertSettingMetadata(List<SettingsMetadata> settingsMetadataList) {
+		try {
+			String insertSettingMetadata = "INSERT INTO core.settings_metadata ( settings_id, document_id, identifier, "
+					+ "server_version, team, team_id, provider_id, location_id, uuid, json, setting_type, setting_value, setting_key, setting_description, setting_label, inherited_from) VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)";
+
+			Connection connection = null;
+
+			try {
+				connection = DataSourceUtils.getConnection(openSRPDataSource);
+				PreparedStatement preparedStatement = connection.prepareStatement(insertSettingMetadata);
+				final int batchSize = 1000;
+				int count = 0;
+				for (SettingsMetadata settingsMetadata : settingsMetadataList) {
+
+					ObjectMapper objectMapper = new ObjectMapper();
+					String json = objectMapper.writeValueAsString(settingsMetadata.getJson());
+
+					PGobject pGobject = new PGobject();
+					pGobject.setType("json");
+					pGobject.setValue(json);
+
+					preparedStatement.setLong(1, settingsMetadata.getSettingsId());
+					preparedStatement.setString(2, settingsMetadata.getDocumentId());
+					preparedStatement.setString(3, settingsMetadata.getIdentifier());
+					preparedStatement.setLong(4, settingsMetadata.getServerVersion());
+					preparedStatement.setString(5, settingsMetadata.getTeam());
+					preparedStatement.setString(6, settingsMetadata.getTeamId());
+					preparedStatement.setString(7, settingsMetadata.getProviderId());
+					preparedStatement.setString(8, settingsMetadata.getLocationId());
+					preparedStatement.setString(9, settingsMetadata.getUuid());
+					preparedStatement.setObject(10, pGobject);
+					preparedStatement.setString(11, settingsMetadata.getSettingType());
+					preparedStatement.setString(12, settingsMetadata.getSettingValue());
+					preparedStatement.setString(13, settingsMetadata.getSettingKey());
+					preparedStatement.setString(14, settingsMetadata.getSettingDescription());
+					preparedStatement.setString(15, settingsMetadata.getSettingLabel());
+					preparedStatement.setString(16, settingsMetadata.getInheritedFrom());
+					preparedStatement.addBatch();
+					if (++count % batchSize == 0) {
+						preparedStatement.executeBatch();
+					}
```

Comment:
When using prepared statements is this necessary? You should execute the statement once done binding all params

Reply:
It might come in handy in case we have more items on the payload.  It will help us avoid `java.lang.OutOfMemoryError: Java heap space` but this is only for very large datasets.